[tohpinren] iP

tohpinren opened the pull request:
# Add Week 2 tasks (Level-0 to Level-6):

> “A man who does not plan long ahead will find trouble at his door.” ― Confucius, Chinese philosopher

Anto (Duke) helps you **remember** things you need to do. It's:
- text-based
- easy to learn
- SUPER FAST to use

## All you need to do is:
1. Download it [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
2. Double-click it.
3. Add your tasks. `todo FirstTask`
4. Let it manage your tasks for you 😄
And it is FREE!
```
===---===---===---===---===---===---===---===
Hello I'm Anto
What can I do for you?
===---===---===---===---===---===---===---===
```

## Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

## What is in this PR?
1. Level 0: Rename, Greet, Exit
2. Level 1: Echo
3. Level-2: Add, List
4. Level-3: Mark as Done
6. Level-4: ToDo, Event, Deadline
7. A-TextUiTesting: Automated Text UI Testing
8. Level-5: Handle Errors
9. Level-6: Delete

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

tohpinren committed:
Level-0: Abstract printing, rename Duke to Anto

tohpinren committed:
Level-1

tohpinren committed:
Level-2

tohpinren committed:
Level-3

tohpinren committed:
Level-4

tohpinren committed:
A-TextUiTesting

tohpinren committed:
Level-5

tohpinren committed:
Level-6

tohpinren committed:
Fix Exceptions

tohpinren committed:
Level-7

tohpinren committed:
Level-7

tohpinren committed:
Level-8

tohpinren committed:
Level-8

tohpinren committed:
A-MoreOOP

tohpinren committed:
A-MoreOOP

tohpinren committed:
A-Packages: Move classes into anto package

Classes are not organised into packages currently.

Packages can help make the codebase more organised as it grows.

Created anto package and move classes into it.

tohpinren committed:
A-Packages

tohpinren committed:
Merge branch 'add-gradle-support'

tohpinren committed:
A-Gradle: Add gradle

Project does not currently use gradle

Gradle can be used to automate build tasks

Add gradle to project

tohpinren committed:
A-JUnit

Project does not have JUnit tests.

We should use JUnit to automate tests.

Add JUnit tests to project

tohpinren committed:
A-JUnit

tohpinren committed:
A-JavaDoc

The project did not have javadocs.

Javadocs can make the code written more clear and improves
collaboration.

Add javadocs to project.

tohpinren committed:
A-JavaDoc

tohpinren committed:
A-CodingStandard

Project did not follow coding standards 100%.

Coding Standards can make the code more readable.

Refactor code using coding standards.

tohpinren committed:
A-CodingStandard

tohpinren committed:
Level-9

There was no way to find specific task in the chatbot.

A find function would be helpful for users to look for specific tasks in
their list.

Add a find function to filter for keyword and print list to user.

tohpinren committed:
Level-9

mingyuanc commented on src/main/java/anto/Anto.java:
```diff
@@ -0,0 +1,47 @@
+     *
+     * @param filePath The relative file path to the location of anto.txt file.
+     */
+    public Anto(String filePath) {
```
Good that your public methods all have javadocs!

mingyuanc commented on src/main/java/anto/Deadline.java:
```diff
@@ -0,0 +1,45 @@
+
+    /**
+     * Creates a Deadline task with specified description and by date.
+     * @param description Description of the deadline task.
```
Should there be an empty line between between description and parameter section?

mingyuanc commented on src/main/java/anto/Parser.java:
```diff
@@ -0,0 +1,206 @@
+
+    /**
+     * Creates a Parser class.
+     * @param ui Ui class that handles printing to command line.
```
Likewise, should there be an empty line here between description and parameter section?

mingyuanc commented on src/main/java/anto/Parser.java:
```diff
@@ -0,0 +1,206 @@
+                    if (input.length() < 6 || input.split(" ", 2).length < 2) {
+                        throw new InvalidIndexException("OOPS!!! You need to include an index to mark task.");
+                    }
+
```
Good that all your logical units within a block are separated by one blank line.

mingyuanc commented on src/main/java/anto/Parser.java:
```diff
@@ -0,0 +1,206 @@
+     * @param strInt String to check.
+     * @return True if string represents an integer.
+     */
+    public static boolean isInt(String strInt) {
```
Good that your method name sounds boolean

mingyuanc commented on src/main/java/anto/Parser.java:
```diff
@@ -0,0 +1,206 @@
+
+    /**
+     * Check if String given is in correct datetime format.
+     * @param dateStr String to be checked.
```
Likewise, should there be a space here?

mingyuanc commented on src/main/java/anto/Storage.java:
```diff
@@ -0,0 +1,78 @@
+
+    /**
+     * Creates a Storage class.
+     * @param ui Ui that handles all printing to the command line.
```
Likewise for here, should there be a space here?

mingyuanc commented on src/main/java/anto/Task.java:
```diff
@@ -0,0 +1,56 @@
+    }
+
+    /**
+     * Returns string representing task.
```
Likewise, should there be a new line here?

mingyuanc commented on src/main/java/anto/TaskList.java:
```diff
@@ -0,0 +1,218 @@
+        try {
+            FileWriter writer = new FileWriter(antoFile, true);
+            if (newTask instanceof Todo) {
+                writer.write(String.format("\nT | %s | %s", newTask.isDone ? "1" : "0", newTask.description));
```
Should this be encapsulated in each task's implementation instead of here? for example in a method called getStoreFormat() for example

mingyuanc commented on src/main/java/anto/TaskList.java:
```diff
@@ -0,0 +1,218 @@
+                writer.write(String.format("\nD | %s | %s | %s", newTask.isDone ? "1" : "0", newTask.description, (
+                        (Deadline) newTask).by.format(DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm"))));
+            } else if (newTask instanceof Event) {
+                writer.write(String.format("\nE | %s | %s | %s | %s", newTask.isDone ? "1" : "0", newTask.description, (
```
Should the line break be after the comma instead of after the bracket?

mingyuanc commented on src/main/java/anto/TaskList.java:
```diff
@@ -0,0 +1,218 @@
+
+            int line = 0;
+
+            while (sc.hasNextLine()) {
```
Should all this logic be placed in Storage class instead? Since it is handling reading and writing to storage.

mingyuanc commented on src/main/java/anto/TaskList.java:
```diff
@@ -0,0 +1,218 @@
+        File tempFile = tempAbsolutePath.toFile();
+
+        try {
+            Scanner sc = new Scanner(antoFile);
```
Should all this logic be placed in Storage class instead? Since it is handling reading and writing to storage.

mingyuanc commented on src/main/java/anto/TaskList.java:
```diff
@@ -0,0 +1,218 @@
+        Path tempAbsolutePath = Paths.get(tempRelativePath).toAbsolutePath();
+        File tempFile = tempAbsolutePath.toFile();
+
+        try {
```
Should all this logic be placed in Storage class instead? Since it is handling reading and writing to storage.

mingyuanc commented on src/main/java/anto/TaskList.java:
```diff
@@ -0,0 +1,218 @@
+     *
+     * @return ArrayList of tasks.
+     */
+    public ArrayList<Task> getStorage() {
```
should this be named as something else? Could be mistaken for returning a Storage instance

mingyuanc reviewed (commented):
Overall your coding standard is pretty good!
Feel that some logic could be encapsulated better in other classes

tohpinren committed:
A-CheckStyle

Styling can be improved.

Styling can improve readability of code.

Improve styling using checkstyle and PR comments.

tohpinren committed:
A-CheckStyle

tohpinren committed:
Level-10

Chatbot ran on CLI, did not have a GUI.

GUI can make user experience better.

Add a GUI for Chatbot using JavaFX.

tohpinren committed:
Level-10

tohpinren committed:
Level-10

Project did not have a GUI.

GUI can improve user experience.

Add a GUI using JavaFX.

tohpinren committed:
Level-10

seraphimstreets commented on src/main/java/anto/Ui.java:
```diff
@@ -0,0 +1,192 @@
+    /**
+     * Creates a Ui class.
+     */
+    public Ui() {}
```
Perhaps this could be omitted, sine the default constructor does nothing? I think this might not be Egyptian brackets as well.

seraphimstreets commented on src/main/java/anto/Task.java:
```diff
@@ -0,0 +1,58 @@
+    /**
+     * Boolean representing whether task is completed.
+     */
+    protected boolean isDone;
```
Is there a reason why these variables are protected instead of private? For instance, you have setter methods which modify isDone, but they are all used in the same package, so are redundant since there is already direct access to these varaibles anyway.

seraphimstreets commented on src/main/java/anto/Ui.java:
```diff
@@ -0,0 +1,192 @@
+                + "What can I do for you?";
+    }
+
+    /**
```
Description could probably be more accurate, like the string that leaves user a parting message, since it doesn't just say bye.

seraphimstreets commented on src/main/java/anto/MainWindow.java:
```diff
@@ -0,0 +1,78 @@
+
+    /**
+     * Creates two dialog boxes, one echoing user input and the other containing Anto's reply and then appends them to
+     * the dialog container. Clears the user input after processing. Ends if Anto response is bye message.
```
Should be ends if user response is bye message?

seraphimstreets commented on src/main/java/anto/Storage.java:
```diff
@@ -0,0 +1,214 @@
+        try {
+            FileWriter writer = new FileWriter(antoFile, true);
+            if (task instanceof Todo) {
+                // It is safe to type cast because the type is checked before
```
I like these comments that highlight typecasting safety. Although maybe they could be condensed into 1 comment at the top?

seraphimstreets reviewed (commented):
All in all, I liked how you structured your code in a clean and understandable manner, and documented everything clearly. Most of my comments are just minor nitpicks and this would be quite a pleasant codebase to read and work on.

tohpinren committed:
Add assertions to the project

Assertions are not used for the Anto Chatbot.

Assertions can help us ensure that the important assumptions at various
points of the code are held true.

Let's add assert statements into various parts of the code to check that
our assumptions are true.

We need to ensure our assumptions about the code is true at various
points and we can use assert statements to do so.

tohpinren committed:
Refactor to improve code quality

Code quality is not up to standard.

Let's use what I learnt from the Code Quality topics to refactor my code
to improve the overall code quality.

Having better code quality improves readability and allows for better
maintenance and improvements in the future.

tohpinren committed:
Merge pull request #1 from tohpinren/branch-A-Assertions

Add assertions to the project

tohpinren committed:
Merge branch 'master' into branch-A-CodeQuality

tohpinren committed:
Merge pull request #2 from tohpinren/branch-A-CodeQuality

Refactor to improve code quality

tohpinren committed:
Set up GitHub Actions CI

Project is not using GitHub Actions.

GitHub Actions CI can improve workflow by automating steps.

Let's add GitHub Actions CI to the project.

tohpinren committed:
Merge pull request #3 from tohpinren/branch-A-CI

Set up GitHub Actions CI

tohpinren committed:
Add feature to view schedules on given date.

Users cannot look up the events happening on a given date.

With view schedule function, users can easily look up the events
happening on a date.

Let's add the view schedule funciton to the Anto Chatbot.

tohpinren committed:
Merge pull request #4 from tohpinren/branch-B-ViewSchedules

Add feature to view schedules on given date.

tohpinren committed:
Add User Guide

There was no User Guide.

User Guide will teach a user how to use Anto.

Let's add a user guide.

tohpinren committed:
Merge pull request #6 from tohpinren/branch-A-UserGuide

Add User Guide

tohpinren committed:
Fix formatting issues

Some parts of the product website are not rendered properly.

Let's fix it.

tohpinren committed:
Merge pull request #7 from tohpinren/branch-A-UserGuide

Fix formatting issues

tohpinren committed:
Fix: Table in User Guide

tohpinren committed:
Merge pull request #8 from tohpinren/branch-A-UserGuide

Fix: Table in User Guide

tohpinren committed:
Fix: Remove command summary table

tohpinren committed:
Merge pull request #9 from tohpinren/branch-A-UserGuide

Fix: Remove command summary table

tohpinren committed:
Fix: IO bug and index error

tohpinren committed:
Add Anto to README title

tohpinren committed:
Merge pull request #10 from tohpinren/branch-A-UserGuide

Add Anto to README title
